src/experimental/ProbabilisticGraphicalModels/conditioning.jl

```diff
@@ -7,8 +7,16 @@ function condition(
     bn::BayesianNetwork{V}, conditioning_variables_and_values::Dict{V,<:Any}
 ) where {V}
     is_observed = copy(bn.is_observed)
-    values = copy(bn.values)
-    bn_new = BangBang.setproperties!!(bn; is_observed=is_observed, values=values)
+    evaluation_env = merge(
+        bn.evaluation_env,
+        NamedTuple{Tuple(Symbol(k) for k in keys(conditioning_variables_and_values))}(
```

Comment:
key of `conditioning_variables_and_values` can be VarName, here just use AbstractPPL.set should work